public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,885 @@
+
+  ### When to set form model values (_ngOnChanges_)
+
+  Now you know _how_ to set the _form model_ values. But _when_ do you set them?
+  The answer depends upon when the component gets the _data model_ values.
+  
+  If the `HeroDetailComponent` is editing a _new_ hero, the data model consists entirely of default values
+  that you could copy to the `FormControls` when you create them in the component's constructor.
+  That's effectively what you did when you set each control to the empty string.
+
+  If the `HeroDetailComponent` is editing an _existing_ hero, you have to wait for that hero to arrive.
+  The component _might_ have the ability to retrieve the hero from the server (via an injected data service)
+  in which case you could set the form model in the [ngOnInit](lifecyle-hooks.html#oninit) lifecycle hook.
+  The final version of the `HeroDetailComponent` in the [Tour of Heroes](../tutorial/toh-pt6.html) does it this way.
+
+  The `HeroDetailComponent` in this reactive forms sample works a little differently.
+  It's the _detail_ view nested within a _master/detail_ `HeroListComponent` ([discussed below](#hero-list)).
+  
+  The `HeroListComponent` displays hero names to the user.
+  When the user clicks on a hero, the list component passes the selected hero into the `HeroDetailComponent`
+  by **binding to the latter's `hero` input property**.
+
++makeExample('reactive-forms/ts/app/hero-list.component.1.html', '','hero-list.component.html (simplified)')(format=".")
+
+:marked
+  In this approach, the value of `hero` in the `HeroDetailComponent` changes 
+  every time the user selects a new hero.
+  You can't call _setValue_ in `ngOnInit` or you'll miss the user's selections.
+  Put the _setValue_ logic in the [ngOnChanges](lifecyle-hooks.html#onchanges)
+  hook, which Angular calls whenever the input `hero` property changes.
+
+  You should also _reset the form_ when the hero changes so that 
+  control values from the previous hero are cleared and 
+  status flags are restored to the _pristine_ state.
+
+  Here's what `ngOnChanges` should look like:
+
++makeExample('reactive-forms/ts/app/hero-detail-versions.component.ts', 'set-value-on-changes','app/hero-detail.component.ts (excerpt)')(format=".")
+
+a#hero-list
+:marked
+  ### Create the _HeroListComponent_ and _HeroService_
+
+  The `HeroDetailComponent` is a nested sub-component of the `HeroListComponent` in a _master/detail_ view.
+  Together they look a bit like this:
+
+figure.image-display
+  img(src="/resources/images/devguide/reactive-forms/hero-list.png" width="420px" alt="HeroListComponent")
+:marked
+  The `HeroListComponent` uses an injected `HeroService` to retrieve heroes from the server
+  and then presents those heroes to the user as a series of buttons.
+  The `HeroService` emulates an HTTP service. 
+  It returns a `Promise` of heroes that resolves after a short delay, 
```

Comment:
Because we didn't want to burden this example with RxJS imports.  The http guide shows both techniques.

Incidentally, in the developer community the jury is still out on whether a data service should return an `Observable` or a `Promise` to its consuming component.  Even Ben Lesh is waffling.